testsuite/smoke-test/src/aptos/gas_check.rs

```diff
@@ -35,6 +35,17 @@ impl AptosTest for GasCheck {
         assert!(format!("{:?}", err).contains("INSUFFICIENT_BALANCE_FOR_TRANSACTION_FEE"));
 
         ctx.mint(account1.address(), 1000).await?;
+        ctx.mint(account2.address(), 1000).await?;
+
+        let transfer_too_much =
+            account2.sign_with_transaction_builder(ctx.aptos_transaction_factory().payload(
+                aptos_stdlib::encode_test_coin_transfer(account1.address(), 1000),
+            ));
+
+        ctx.client()
+            .submit_and_wait(&transfer_too_much)
+            .await
+            .unwrap_err();
```

Comment:
Does this guarantee it's an error instead of a discard?

Reply:
discard will timeout this request and probably timeout the whole test

Reply:
K. Would be nice if we can test the explicit error. But this works for me.